Problem linking against VDT in the latest versions

### Check duplicate issues.

- [x] Checked for duplicates

### Description

When compiling [FastFrames](https://gitlab.cern.ch/atlas-amglab/fastframes), am getting the following error during linking:
```
tomas@tomas-IdeaPad-5-Pro-14IAP7:~/RDataFrame/FastFrames/build$ make
[  5%] Built target FastFrames_dict
[  8%] Linking CXX shared library lib/libFastFrames.so
/usr/bin/ld: cannot find -lVDT: No such file or directory
collect2: error: ld returned 1 exit status
```

This started happening since nightly from July 3rd  (July 2nd is fine). 

@will-cern suggested that this PR: https://github.com/root-project/root/pull/8709 might have caused the issue as it matches the timeline

### Reproducer

Compile FastFrames with ROOT master

### ROOT version

```
ROOT Version: 6.37.01
Built for linuxx8664gcc on Jul 08 2025, 12:14:14
From heads/master@v6-37-01-7373-g2a6f2e772d8
```

### Installation method

Built from source and using LCG nightly

### Operating system

Linux

### Additional context

_No response_

--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -1544,9 +1544,7 @@ if(vdt OR builtin_vdt)
           set(builtin_vdt ON CACHE BOOL "Enabled because external vdt not found (${vdt_description})" FORCE)
         endif()
       endif()
-     else()
-       add_library(VDT ALIAS VDT::VDT)
-     endif()
+    endif()
   endif()
   if(builtin_vdt)
     set(vdt_version 0.4.6)
@@ -1577,25 +1575,17 @@ if(vdt OR builtin_vdt)
     )
     set(VDT_INCLUDE_DIR ${CMAKE_BINARY_DIR}/ginclude)
     set(VDT_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/ginclude)
-    if(NOT TARGET VDT)
-      add_library(VDT IMPORTED SHARED)
-      add_dependencies(VDT BUILTIN_VDT)
-      set_target_properties(VDT PROPERTIES IMPORTED_LOCATION "${VDT_LIBRARIES}")
-      target_include_directories(VDT INTERFACE $<BUILD_INTERFACE:${VDT_INCLUDE_DIR}> $<INSTALL_INTERFACE:include/>)
-    endif()
 
-    install(FILES ${CMAKE_BINARY_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}vdt${CMAKE_SHARED_LIBRARY_SUFFIX}
+    add_library(VDT::VDT SHARED IMPORTED GLOBAL)
+    add_dependencies(VDT::VDT BUILTIN_VDT)
+    set_target_properties(VDT::VDT PROPERTIES IMPORTED_LOCATION "${VDT_LIBRARIES}")
+    target_include_directories(VDT::VDT INTERFACE $<BUILD_INTERFACE:${VDT_INCLUDE_DIR}> $<INSTALL_INTERFACE:include/>)
+
+    install(FILES ${VDT_LIBRARIES}
             DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libraries)
     install(DIRECTORY ${CMAKE_BINARY_DIR}/include/vdt
             DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT extra-headers)
-    set(vdt ON CACHE BOOL "Enabled because builtin_vdt enabled (${vdt_description})" FORCE)
-    set_property(GLOBAL APPEND PROPERTY ROOT_BUILTIN_TARGETS VDT)
-    add_library(VDT::VDT STATIC IMPORTED GLOBAL)
-    set_target_properties(VDT::VDT
-      PROPERTIES
-        IMPORTED_LOCATION "${VDT_LIBRARIES}"
-        INTERFACE_INCLUDE_DIRECTORIES "${VDT_INCLUDE_DIRS}"
-    )
+    set_property(GLOBAL APPEND PROPERTY ROOT_BUILTIN_TARGETS VDT::VDT)
   endif()
 endif()
 

--- a/math/vecops/CMakeLists.txt
+++ b/math/vecops/CMakeLists.txt
@@ -20,7 +20,7 @@ ROOT_STANDARD_LIBRARY_PACKAGE(ROOTVecOps
 )
 
 if(builtin_vdt OR vdt)
-  target_link_libraries(ROOTVecOps PUBLIC VDT)
+  target_link_libraries(ROOTVecOps PUBLIC VDT::VDT)
 endif()
 
 include(CheckCXXSymbolExists)

--- a/roofit/batchcompute/CMakeLists.txt
+++ b/roofit/batchcompute/CMakeLists.txt
@@ -10,7 +10,7 @@ ROOT_LINKER_LIBRARY(RooBatchCompute
 target_include_directories(RooBatchCompute PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/res>)
 
 if(vdt OR builtin_vdt)
-  target_link_libraries(RooBatchCompute PUBLIC VDT)
+  target_link_libraries(RooBatchCompute PRIVATE VDT::VDT)
 endif()
 
 

--- a/tmva/tmva/CMakeLists.txt
+++ b/tmva/tmva/CMakeLists.txt
@@ -405,7 +405,7 @@ if(MSVC)
 endif()
 
 if(vdt OR builtin_vdt)
-  target_link_libraries(TMVA PRIVATE VDT)
+  target_link_libraries(TMVA PRIVATE VDT::VDT)
 endif()
 
 if(tmva-cpu)

